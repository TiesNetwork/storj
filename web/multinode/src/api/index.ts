// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * ErrorUnauthorized is a custom error type for performing unauthorized operations.
 */
export class UnauthorizedError extends Error {
    public constructor(message: string = 'authorization required') {
        super(message);
    }
}

/**
 * BadRequestError is a custom error type for performing bad request.
 */
export class BadRequestError extends Error {
    public constructor(message: string = 'bad request') {
        super(message);
    }
}

/**
 * InternalError is a custom error type for internal server error.
 */
export class InternalError extends Error {
    public constructor(message: string = 'internal server error') {
        super(message);
    }
}
