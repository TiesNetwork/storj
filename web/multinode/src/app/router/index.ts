// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Router, { RouterMode } from 'vue-router';
import { Component } from 'vue-router/types/router';

import AddFirstNode from '@/app/views/AddFirstNode.vue';
import Dashboard from '@/app/views/Dashboard.vue';
import MyNodes from '@/app/views/MyNodes.vue';
import WelcomeScreen from '@/app/views/WelcomeScreen.vue';

/**
 * Metadata holds arbitrary information to routes like transition names, who can access the route, etc.
 */
class Metadata {
    public requiresAuth: boolean;
}

/**
 * Route holds all needed information to fill up router config.
 */
export class Route {
    public readonly path: string;
    public readonly name: string;
    public readonly component: Component;
    public children?: Route[];
    public readonly meta?: Metadata;

    /**
     * default constructor.
     * @param path - route path.
     * @param name - name of the route, needed fot identifying route by name.
     * @param component - component mapped to route.
     * @param children - all nested components of current route.
     * @param meta - arbitrary information to routes like transition names, who can access the route, etc.
     */
    public constructor(path: string, name: string, component: Component, meta: Metadata | undefined = undefined) {
        this.path = path;
        this.name = name;
        this.component = component;
        this.meta = meta;
    }

    /**
     * Adds children routes to route.
     */
    public addChildren(children: Route[]): Route {
        this.children = children;

        return this;
    }
}

/**
 * Config contains configuration of all available routes for a Multinode Dashboard router.
 */
export class Config {
    public static Root: Route = new Route('/', 'Root', Dashboard, {requiresAuth: true});
    public static Welcome: Route = new Route('/welcome', 'Welcome', WelcomeScreen);
    public static AddFirstNode: Route = new Route('/add-first-node', 'AddFirstNode', AddFirstNode);
    public static MyNodes: Route = new Route('/my-nodes', 'MyNodes', MyNodes);

    public static mode: RouterMode = 'history';
    public static routes: Route[] = [
        Config.Root.addChildren([
            Config.MyNodes,
        ]),
        Config.Welcome,
        Config.AddFirstNode,
    ];
}

export const router = new Router(Config);
