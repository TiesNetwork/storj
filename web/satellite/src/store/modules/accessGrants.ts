// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/store';
import {
    AccessGrant,
    AccessGrantCursor,
    AccessGrantsApi,
    AccessGrantsOrderBy,
    AccessGrantsPage,
    DurationPermission,
    GatewayCredentials,
} from '@/types/accessGrants';
import { SortDirection } from '@/types/common';

export const ACCESS_GRANTS_ACTIONS = {
    FETCH: 'fetchAccessGrants',
    CREATE: 'createAccessGrant',
    DELETE: 'deleteAccessGrants',
    CLEAR: 'clearAccessGrants',
    GET_GATEWAY_CREDENTIALS: 'getGatewayCredentials',
    SET_ACCESS_GRANTS_WEB_WORKER: 'setAccessGrantsWebWorker',
    STOP_ACCESS_GRANTS_WEB_WORKER: 'stopAccessGrantsWebWorker',
    SET_SEARCH_QUERY: 'setAccessGrantsSearchQuery',
    SET_SORT_BY: 'setAccessGrantsSortingBy',
    SET_SORT_DIRECTION: 'setAccessGrantsSortingDirection',
    SET_DURATION_PERMISSION: 'setAccessGrantsDurationPermission',
    TOGGLE_SELECTION: 'toggleAccessGrantsSelection',
    TOGGLE_BUCKET_SELECTION: 'toggleBucketSelection',
    CLEAR_SELECTION: 'clearAccessGrantsSelection',
};

export const ACCESS_GRANTS_MUTATIONS = {
    SET_PAGE: 'setAccessGrants',
    SET_GATEWAY_CREDENTIALS: 'setGatewayCredentials',
    SET_ACCESS_GRANTS_WEB_WORKER: 'setAccessGrantsWebWorker',
    STOP_ACCESS_GRANTS_WEB_WORKER: 'stopAccessGrantsWebWorker',
    TOGGLE_SELECTION: 'toggleAccessGrantsSelection',
    TOGGLE_BUCKET_SELECTION: 'toggleBucketSelection',
    CLEAR_SELECTION: 'clearAccessGrantsSelection',
    CLEAR: 'clearAccessGrants',
    CHANGE_SORT_ORDER: 'changeAccessGrantsSortOrder',
    CHANGE_SORT_ORDER_DIRECTION: 'changeAccessGrantsSortOrderDirection',
    SET_SEARCH_QUERY: 'setAccessGrantsSearchQuery',
    SET_PAGE_NUMBER: 'setAccessGrantsPage',
    SET_DURATION_PERMISSION: 'setAccessGrantsDurationPermission',
};

const {
    SET_PAGE,
    TOGGLE_SELECTION,
    TOGGLE_BUCKET_SELECTION,
    CLEAR_SELECTION,
    CLEAR,
    CHANGE_SORT_ORDER,
    CHANGE_SORT_ORDER_DIRECTION,
    SET_SEARCH_QUERY,
    SET_PAGE_NUMBER,
    SET_DURATION_PERMISSION,
    SET_GATEWAY_CREDENTIALS,
    SET_ACCESS_GRANTS_WEB_WORKER,
    STOP_ACCESS_GRANTS_WEB_WORKER,
} = ACCESS_GRANTS_MUTATIONS;

export class AccessGrantsState {
    public cursor: AccessGrantCursor = new AccessGrantCursor();
    public page: AccessGrantsPage = new AccessGrantsPage();
    public selectedAccessGrantsIds: string[] = [];
    public selectedBucketNames: string[] = [];
    public permissionNotBefore: Date = new Date();
    public permissionNotAfter: Date = new Date('2200-01-01');
    public gatewayCredentials: GatewayCredentials = new GatewayCredentials();
    public accessGrantsWebWorker: Worker | null = null;
    public isAccessGrantsWebWorkerReady: boolean = false;
}

/**
 * creates access grants module with all dependencies
 *
 * @param api - accessGrants api
 */
export function makeAccessGrantsModule(api: AccessGrantsApi): StoreModule<AccessGrantsState> {
    return {
        state: new AccessGrantsState(),
        mutations: {
            [SET_ACCESS_GRANTS_WEB_WORKER](state: AccessGrantsState): void {
                state.accessGrantsWebWorker = new Worker('@/../static/wasm/accessGrant.worker.js', { type: 'module' });
                state.accessGrantsWebWorker.onmessage = (event: MessageEvent) => {
                    const data = event.data;
                    if (data !== 'configured') {
                        console.error('Failed to configure access grants web worker');

                        return;
                    }

                    state.isAccessGrantsWebWorkerReady = true;
                };
                state.accessGrantsWebWorker.onerror = (error: ErrorEvent) => {
                    console.error(`Failed to configure access grants web worker. ${error.message}`);
                };
            },
            [STOP_ACCESS_GRANTS_WEB_WORKER](state: AccessGrantsState): void {
                state.accessGrantsWebWorker?.postMessage({
                    'type': 'Stop',
                });
                state.accessGrantsWebWorker = null;
                state.isAccessGrantsWebWorkerReady = false;
            },
            [SET_PAGE](state: AccessGrantsState, page: AccessGrantsPage) {
                state.page = page;
                state.page.accessGrants = state.page.accessGrants.map(accessGrant => {
                    if (state.selectedAccessGrantsIds.includes(accessGrant.id)) {
                        accessGrant.isSelected = true;
                    }

                    return accessGrant;
                });
            },
            [SET_GATEWAY_CREDENTIALS](state: AccessGrantsState, credentials: GatewayCredentials) {
                state.gatewayCredentials = credentials;
            },
            [SET_PAGE_NUMBER](state: AccessGrantsState, pageNumber: number) {
                state.cursor.page = pageNumber;
            },
            [SET_SEARCH_QUERY](state: AccessGrantsState, search: string) {
                state.cursor.search = search;
            },
            [SET_DURATION_PERMISSION](state: AccessGrantsState, permission: DurationPermission) {
                state.permissionNotBefore = permission.notBefore;
                state.permissionNotAfter = permission.notAfter;
            },
            [CHANGE_SORT_ORDER](state: AccessGrantsState, order: AccessGrantsOrderBy) {
                state.cursor.order = order;
            },
            [CHANGE_SORT_ORDER_DIRECTION](state: AccessGrantsState, direction: SortDirection) {
                state.cursor.orderDirection = direction;
            },
            [TOGGLE_SELECTION](state: AccessGrantsState, accessGrant: AccessGrant) {
                if (!state.selectedAccessGrantsIds.includes(accessGrant.id)) {
                    state.page.accessGrants.forEach((grant: AccessGrant) => {
                        if (grant.id === accessGrant.id) {
                            grant.isSelected = true;
                        }
                    });
                    state.selectedAccessGrantsIds.push(accessGrant.id);

                    return;
                }

                state.page.accessGrants.forEach((grant: AccessGrant) => {
                    if (grant.id === accessGrant.id) {
                        grant.isSelected = false;
                    }
                });
                state.selectedAccessGrantsIds = state.selectedAccessGrantsIds.filter(accessGrantId => {
                    return accessGrant.id !== accessGrantId;
                });
            },
            [TOGGLE_BUCKET_SELECTION](state: AccessGrantsState, bucketName: string) {
                if (!state.selectedBucketNames.includes(bucketName)) {
                    state.selectedBucketNames.push(bucketName);

                    return;
                }

                state.selectedBucketNames = state.selectedBucketNames.filter(name => {
                    return bucketName !== name;
                });
            },
            [CLEAR_SELECTION](state: AccessGrantsState) {
                state.selectedBucketNames = [];
                state.selectedAccessGrantsIds = [];
                state.page.accessGrants = state.page.accessGrants.map((accessGrant: AccessGrant) => {
                    accessGrant.isSelected = false;

                    return accessGrant;
                });
            },
            [CLEAR](state: AccessGrantsState) {
                state.cursor = new AccessGrantCursor();
                state.page = new AccessGrantsPage();
                state.selectedAccessGrantsIds = [];
                state.selectedBucketNames = [];
                state.permissionNotBefore = new Date();
                state.permissionNotAfter = new Date('2200-01-01');
                state.gatewayCredentials = new GatewayCredentials();
            },
        },
        actions: {
            setAccessGrantsWebWorker: function({commit}: any): void {
                commit(SET_ACCESS_GRANTS_WEB_WORKER);
            },
            stopAccessGrantsWebWorker: function({commit}: any): void {
                commit(STOP_ACCESS_GRANTS_WEB_WORKER);
            },
            fetchAccessGrants: async function ({commit, rootGetters, state}, pageNumber: number): Promise<AccessGrantsPage> {
                const projectId = rootGetters.selectedProject.id;
                commit(SET_PAGE_NUMBER, pageNumber);

                const accessGrantsPage: AccessGrantsPage = await api.get(projectId, state.cursor);
                commit(SET_PAGE, accessGrantsPage);

                return accessGrantsPage;
            },
            createAccessGrant: async function ({commit, rootGetters}: any, name: string): Promise<AccessGrant> {
                const accessGrant = await api.create(rootGetters.selectedProject.id, name);

                return accessGrant;
            },
            deleteAccessGrants: async function({state}: any): Promise<void> {
                await api.delete(state.selectedAccessGrantsIds);
            },
            getGatewayCredentials: async function({state, commit}: any, accessGrant: string): Promise<void> {
                const credentials: GatewayCredentials = await api.getGatewayCredentials(accessGrant);

                commit(SET_GATEWAY_CREDENTIALS, credentials);
            },
            setAccessGrantsSearchQuery: function ({commit}, search: string) {
                commit(SET_SEARCH_QUERY, search);
            },
            setAccessGrantsSortingBy: function ({commit}, order: AccessGrantsOrderBy) {
                commit(CHANGE_SORT_ORDER, order);
            },
            setAccessGrantsSortingDirection: function ({commit}, direction: SortDirection) {
                commit(CHANGE_SORT_ORDER_DIRECTION, direction);
            },
            setAccessGrantsDurationPermission: function ({commit}, permission: DurationPermission) {
                commit(SET_DURATION_PERMISSION, permission);
            },
            toggleAccessGrantsSelection: function ({commit}, accessGrant: AccessGrant): void {
                commit(TOGGLE_SELECTION, accessGrant);
            },
            toggleBucketSelection: function ({commit}, bucketName: string): void {
                commit(TOGGLE_BUCKET_SELECTION, bucketName);
            },
            clearAccessGrantsSelection: function ({commit}): void {
                commit(CLEAR_SELECTION);
            },
            clearAccessGrants: function ({commit}): void {
                commit(CLEAR);
                commit(CLEAR_SELECTION);
            },
        },
        getters: {
            selectedAccessGrants: (state: AccessGrantsState) => state.page.accessGrants.filter((grant: AccessGrant) => grant.isSelected),
        },
    };
}
