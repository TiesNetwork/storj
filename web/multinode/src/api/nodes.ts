// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { BadRequestError, InternalError, UnauthorizedError } from '@/api/index';
import { CreateNodeFields, Node, NodeURL } from '@/nodes';
import { HttpClient } from '@/private/http/client';

/**
 * client for nodes controller of MND api.
 */
export class NodesClient {
    private readonly http: HttpClient = new HttpClient();
    private readonly ROOT_PATH: string = '/api/v0/nodes';

    /**
     * handles node addition.
     *
     * @param node - node to add.
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async add(node: CreateNodeFields): Promise<void> {
        const path = `${this.ROOT_PATH}`;
        const response = await this.http.post(path, JSON.stringify(node));

        if (!response.ok) {
            await this.handleError(response);
        }
    }

    /**
     * returns list of node infos.
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async list(): Promise<Node[]> {
        const path = `${this.ROOT_PATH}/infos`;
        const response = await this.http.get(path);

        if (!response.ok) {
            await this.handleError(response);
        }

        const nodeListJson = await response.json();

        return nodeListJson.map(node => new Node(
            node.id,
            node.name,
            node.version,
            new Date(node.lastContact),
            node.diskSpaceUsed,
            node.diskSpaceLeft,
            node.bandwidthUsed,
            0,
            0,
            0,
            node.totalEarned,
        ));
    }

    /**
     * returns list of node infos by satellite.
     *
     * @param satelliteId - id of the satellite.
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async listBySatellite(satelliteId: string): Promise<Node[]> {
        const path = `${this.ROOT_PATH}/infos/${satelliteId}`;
        const response = await this.http.get(path);

        if (!response.ok) {
            await this.handleError(response);
        }

        const nodeListJson = await response.json();

        return nodeListJson.map(node => new Node(
            node.id,
            node.name,
            node.version,
            new Date(node.lastContact),
            0,
            0,
            0,
            node.onlineScore,
            node.auditScore,
            node.suspensionScore,
            node.totalEarned,
        ));
    }

    /**
     * updates nodes name.
     *
     * @param id - id of the node.
     * @param name - new node name.
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async updateName(id: string, name: string): Promise<void> {
        const path = `${this.ROOT_PATH}/${id}`;
        const response = await this.http.patch(path, JSON.stringify({name: name}));

        if (!response.ok) {
            await this.handleError(response);
        }
    }

    /**
     * deletes node.
     *
     * @param id - id of the node.
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async delete(id: string): Promise<void> {
        const path = `${this.ROOT_PATH}/${id}`;
        const response = await this.http.delete(path);

        if (!response.ok) {
            await this.handleError(response);
        }
    }

    /**
     * retrieves list of trusted satellites node urls for a node.
     */
    public async trustedSatellites(): Promise<NodeURL[]> {
        const path = `${this.ROOT_PATH}/trusted-satellites`;
        const response = await this.http.get(path);

        if (!response.ok) {
            await this.handleError(response);
        }

        const urlListJson = await response.json();

        return urlListJson.map(url => new NodeURL(
            url.ID,
            url.Name,
        ));
    }

    /**
     * handles error due to response code.
     * @param response - response from server.
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid ISBN number.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the ISBN number is valid, but no such book exists in the catalog.
     *
     * @throws {@link InternalError}
     * Thrown if the ISBN number is valid, but no such book exists in the catalog.
     *
     * @private
     */
    private async handleError(response: Response): Promise<void> {
        const body = await response.json();

        switch (response.status) {
            case 401: throw new UnauthorizedError(body.error);
            case 400: throw new BadRequestError(body.error);
            case 500:
            default:
                throw new InternalError(body.error);
        }
    }
}
