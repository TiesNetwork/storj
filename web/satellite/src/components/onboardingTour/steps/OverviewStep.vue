// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template src="./overviewStep.html"></template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import VButton from '@/components/common/VButton.vue';

import { RouteConfig } from '@/router';
import { ACCESS_GRANTS_ACTIONS } from '@/store/modules/accessGrants';
import { BUCKET_ACTIONS } from '@/store/modules/buckets';
import { PAYMENTS_ACTIONS } from '@/store/modules/payments';
import { PROJECTS_ACTIONS } from '@/store/modules/projects';
import { ProjectFields } from '@/types/projects';
import { PM_ACTIONS } from '@/utils/constants/actionNames';
import { SegmentEvent } from '@/utils/constants/analyticsEventNames';

@Component({
    components: {
        VButton,
    },
})
export default class OverviewStep extends Vue {
    public isLoading: boolean = false;

    /**
     * Lifecycle hook after initial render.
     * Sets area to needed state.
     */
    public mounted(): void {
        if (this.userHasProject || this.$store.state.paymentsModule.creditCards.length > 0) {
            this.$router.push(RouteConfig.OnboardingTour.with(RouteConfig.AccessGrant).path);

            return;
        }

        if (this.$store.getters.isTransactionProcessing || this.$store.getters.isBalancePositive) {
            this.$router.push(RouteConfig.OnboardingTour.with(RouteConfig.PaymentStep).path);
        }
    }

    /**
     * Holds button click logic.
     * Redirects to next step (adding payment method).
     */
    public onAddPaymentClick(): void {
        this.$router.push(RouteConfig.OnboardingTour.with(RouteConfig.PaymentStep).path);
    }

    /**
     * Holds button click logic.
     * Creates untitled project and redirects to next step (creating access grant).
     */
    public async onCreateGrantClick(): Promise<void> {
        if (this.isLoading) return;

        this.isLoading = true;

        try {
            const FIRST_PAGE = 1;
            const UNTITLED_PROJECT_NAME = 'Untitled Project';
            const UNTITLED_PROJECT_DESCRIPTION = '___';
            const project = new ProjectFields(
                UNTITLED_PROJECT_NAME,
                UNTITLED_PROJECT_DESCRIPTION,
                this.$store.getters.user.id,
            );
            const createdProject = await this.$store.dispatch(PROJECTS_ACTIONS.CREATE, project);
            const createdProjectId = createdProject.id;

            this.$segment.track(SegmentEvent.PROJECT_CREATED, {
                project_id: createdProjectId,
            });

            await this.$store.dispatch(PROJECTS_ACTIONS.SELECT, createdProjectId);
            await this.$store.dispatch(PM_ACTIONS.CLEAR);
            await this.$store.dispatch(PM_ACTIONS.FETCH, FIRST_PAGE);
            await this.$store.dispatch(PAYMENTS_ACTIONS.GET_PAYMENTS_HISTORY);
            await this.$store.dispatch(PAYMENTS_ACTIONS.GET_BALANCE);
            await this.$store.dispatch(PAYMENTS_ACTIONS.GET_PROJECT_USAGE_AND_CHARGES_CURRENT_ROLLUP);
            await this.$store.dispatch(PROJECTS_ACTIONS.GET_LIMITS, createdProjectId);
            await this.$store.dispatch(ACCESS_GRANTS_ACTIONS.CLEAR);
            await this.$store.dispatch(BUCKET_ACTIONS.CLEAR);

            this.isLoading = false;

            await this.$router.push(RouteConfig.OnboardingTour.with(RouteConfig.AccessGrant).with(RouteConfig.AccessGrantName).path);
        } catch (error) {
            await this.$notify.error(error.message);
            this.isLoading = false;
        }
    }

    /**
     * Indicates if paywall is enabled.
     */
    public get isPaywallEnabled(): boolean {
        return this.$store.state.paymentsModule.isPaywallEnabled;
    }

    /**
     * Indicates if user has at least one project.
     */
    private get userHasProject(): boolean {
        return this.$store.state.projectsModule.projects.length > 0;
    }
}
</script>

<style scoped lang="scss" src="./overviewStep.scss"></style>
