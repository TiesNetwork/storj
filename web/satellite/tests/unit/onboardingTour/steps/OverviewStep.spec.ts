// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import OverviewStep from '@/components/onboardingTour/steps/OverviewStep.vue';

import { PaymentsHttpApi } from '@/api/payments';
import { router } from '@/router';
import { makePaymentsModule, PAYMENTS_MUTATIONS } from '@/store/modules/payments';
import { makeProjectsModule } from '@/store/modules/projects';
import { createLocalVue, mount } from '@vue/test-utils';

import { ProjectsApiMock } from '../../mock/api/projects';

const localVue = createLocalVue();
localVue.use(Vuex);
const projectsApi = new ProjectsApiMock();
const projectsModule = makeProjectsModule(projectsApi);
const paymentsApi = new PaymentsHttpApi();
const paymentsModule = makePaymentsModule(paymentsApi);

const store = new Vuex.Store({ modules: { projectsModule, paymentsModule }});

describe('OverviewStep.vue', (): void => {
    it('renders correctly', async (): Promise<void> => {
        const wrapper = mount(OverviewStep, {
            localVue,
            router,
            store,
        });

        await store.commit(PAYMENTS_MUTATIONS.SET_PAYWALL_ENABLED_STATUS, true);

        expect(wrapper).toMatchSnapshot();

        await store.commit(PAYMENTS_MUTATIONS.SET_PAYWALL_ENABLED_STATUS, false);

        expect(wrapper).toMatchSnapshot();
    });
});
