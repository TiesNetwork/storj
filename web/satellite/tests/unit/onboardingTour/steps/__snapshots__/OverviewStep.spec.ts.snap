// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewStep.vue renders correctly 1`] = `
<div class="overview-area">
  <h1 class="overview-area__title">
    Welcome to Storj.
    <br>
    Let’s Get Started.
  </h1>
  <p class="overview-area__sub-title">
    Follow the docs to start storing data using method below.
  </p>
  <div class="overview-area__steps-area">
    <div class="overview-area__steps-area__step"><img src="@/../static/images/onboardingTour/cli.png" alt="cli image" class="overview-area__steps-area__step__image">
      <h2 class="overview-area__steps-area__step__title">CLI Tool</h2>
      <p class="overview-area__steps-area__step__info">
        Quickly upload data directly through the command line interface.
      </p> <a href="https://documentation.storj.io/setup/cli" target="_blank" class="overview-area__steps-area__step__link">
        CLI Docs &gt;
      </a>
    </div>
  </div>
  <div class="overview-area__divider"></div>
  <p class="overview-area__next-label">Next</p>
  <div class="overview-area__cta container" style="width: 252px; height: 48px;"><span class="label">Add a Payment Method</span></div>
</div>
`;

exports[`OverviewStep.vue renders correctly 2`] = `
<div class="overview-area">
  <h1 class="overview-area__title">
    Welcome to Storj.
    <br>
    Let’s Get Started.
  </h1>
  <p class="overview-area__sub-title">
    Follow the docs to start storing data using method below.
  </p>
  <div class="overview-area__steps-area">
    <div class="overview-area__steps-area__step"><img src="@/../static/images/onboardingTour/cli.png" alt="cli image" class="overview-area__steps-area__step__image">
      <h2 class="overview-area__steps-area__step__title">CLI Tool</h2>
      <p class="overview-area__steps-area__step__info">
        Quickly upload data directly through the command line interface.
      </p> <a href="https://documentation.storj.io/setup/cli" target="_blank" class="overview-area__steps-area__step__link">
        CLI Docs &gt;
      </a>
    </div>
  </div>
  <div class="overview-area__divider"></div>
  <p class="overview-area__next-label">Next</p>
  <div class="overview-area__cta container" style="width: 252px; height: 48px;"><span class="label">Create an Access Grant</span></div>
</div>
`;
